### ISSUE TITLE
✨ Feat: add Libravatar / Gravatar support

### ISSUE THREAD

**Issue by Millefeuille42** [tracim/tracim#6310]
### Feature description

Add Libravatar / Gravatar support for profile pictures.
This extends to any compatible open avatar format.

### How it works

When serving the avatar for a user:
1. Check if there is a user-provided avatar
2. Check if there is a Libravatar / Gravatar avatar for the user's email (is they have an email)
3a. Fallback to the default avatar generated by Libravatar / Gravatar
3b. Fallback to tracim's generated avatar

Several urls for remote avatar can be provided through the instance settings
If none are provided, don't check for remote avatar
Whether to get remote avatar from Libravatar / Gravatar or tracim is set in the instance settings.

Any compatible remote avatar server can be included. This means the avatar can be fetched at this url:

`<protocol>://<FQDN>/avatar/<md5 hash of the user's email>`

**Comment by Millefeuille42** [comment]
Thanks ! 
As this is not a priority and more of a "I'd like this", I am working on it on my spare time

**Comment by PhilippeAccorsi** [comment]
Maybe you can find dev exemple in jitsi code and bigbluebutton code (if it is necessary).

**Comment by Millefeuille42** [comment]
Yes, good idea! The main difficulty currently is to integrate it to the current avatar flow in tracim. Since they are designed to be cached and constant which is not how gravatar should be used, I have to implement a cache invalidation policy for avatars, which would be daily or something for gravatar and indefinite for current avatars.

So what looked like a simple integration turns out to be a small refactor.
